src/inmanta/ast/statements/call.py

```diff
@@ -40,10 +43,23 @@ class FunctionCall(ReferenceStatement):
         contributes:
     """
 
-    def __init__(self, name: LocatableString, arguments):
-        ReferenceStatement.__init__(self, arguments)
-        self.name = name
-        self.arguments = arguments
+    def __init__(
+        self,
+        name: LocatableString,
+        arguments: List[ExpressionStatement],
+        kwargs: List[Tuple[LocatableString, ExpressionStatement]],
+        wrapped_kwargs: List[WrappedKwargs],
+    ) -> None:
```

Comment:
Ensure that a named argument named 'context' is always rejected.

Reply:
Could you provide some more background on this?

Reply:
when a plugin has an argument called 'context', a special context object is passed in, that provides access to e.g. the connection to the server (and even the compilation context (resolver and scheduler) itself)

Reply:
And why should this only be supplied as a positional argument? Also, wouldn't it be better to check on the `Context` type instead of the parameter name? I think I've also seen it called `ctx` instead of `context`.

Reply:
you are correct, it is recognized by type. 

With positional arguments, it is ignored. (i.e. it is not counted in the positions)

With kwargs, it can not necessarily be ignored.

Reply:
And it can never be supplied via the model (inherently)

Reply:
It is a meta argument then? It gets supplied by the compiler?

Reply:
I don't completely understand what should happen. I see that it doesn't get added to the arguments [here](https://github.com/inmanta/inmanta/blob/958b0f5539b4c56a59445a0b544bc5be1b1424f2/src/inmanta/plugins.py#L214). This means it also won't be a valid argument if it's supplied via kwargs. [Here](https://github.com/inmanta/inmanta/blob/958b0f5539b4c56a59445a0b544bc5be1b1424f2/src/inmanta/plugins.py#L374) however it doesn't get special treatment.

Reply:
for example compiling `std::generate_password("pw_id", context=42)` results in
```
Invalid keyword argument 'generate_password' for 'context' (reported in std::generate_password('pw_id',context=42) (./main.cf:1))
```
Clearly, I swapped the function and argument order in the error message, but apart from that I think this is the desired behaviour, no?

Reply:
it is, please add a testcase for this